app/gui2/public/visualizations/HistogramVisualization.vue

```diff
@@ -0,0 +1,665 @@
+    let paddingX = extremesAndDeltas_.dx / (2 * focus_.zoom)
+    domainX = [focus_.x - paddingX, focus_.x + paddingX]
+  }
+  xScale.value.domain(domainX).range([0, boxWidth.value])
+  d3XAxis.value.call(d3.axisBottom(xScale.value).ticks(width.value / 40))
+})
+
+watchEffect(() => {
+  // Update the y-axis in D3.
+  const yScale_ = yScale.value
+  yScale_.range([boxHeight.value, 0])
+  if (yScale_.domain()[0] === 0) {
+    yScale_.domain([0, maxY.value])
+  }
+  fill.value.domain(yScale_.domain())
+  const yTicks = yScale_.ticks().filter(Number.isInteger)
+  yAxis.value.scale(yScale_).tickValues(yTicks)
+  d3YAxis.value.call(yAxis.value)
+})
+
+function redrawData() {
+  const originalXScale_ = originalXScale.value
+  const xScale_ = xScale.value
+  const yScale_ = yScale.value
+  const boxHeight_ = boxHeight.value
+  const fill_ = fill.value
+  let bins: Bin[] = []
+  if (rawBins.value != null) {
+    bins = rawBins.value.map((length, i) => ({ x0: i, x1: i + 1, length }))
+  } else if (points.value != null) {
+    const dataDomain = originalXScale.value.domain()
+    const histogram = d3
+      .bin()
+      .domain([dataDomain[0] ?? 0, dataDomain[1] ?? 1])
+      .thresholds(originalXScale.value.ticks(binCount.value))
+    bins = histogram(points.value)
+  }
+  // The maximum value MUST NOT be 0, otherwise 0 will be in the middle of the y axis.
+  maxY.value = d3.max(bins, (d) => d.length) || 1
+  updateColorLegend(fill.value)
+  d3Plot.value
+    .selectAll('rect')
+    .data(bins)
+    .join(
+      (enter) => enter.append('rect').attr('x', 1),
+      (update) =>
+        update
+          .attr('transform', (d) => `translate(${xScale_(d.x0 ?? 0)}, ${yScale_(d.length ?? 0)})`)
+          .attr('width', (d) => originalXScale_(d.x1 ?? 0) - originalXScale_(d.x0 ?? 0))
+          .attr('height', (d) => Math.max(0, boxHeight_ - yScale_(d.length ?? 0)))
+          .style('fill', (d) => fill_(d.length ?? 0)),
+    )
+}
+
+/**
+ * Update height of the color legend to match the height of the canvas.
+ * Set up `stop` attributes on color legend gradient to match `colorScale`, so color legend shows correct colors
+ * used by histogram.
+ */
+function updateColorLegend(colorScale: ScaleSequential<string>) {
+  const colorScaleToGradient = (t: number, i: number, n: number[]) => ({
+    offset: `${(100 * i) / n.length}%`,
+    color: colorScale(t),
+  })
+  d3ColorLegendGradient.value
+    .selectAll('stop')
+    .data(colorScale.ticks().map(colorScaleToGradient))
+    .enter()
+    .append('stop')
+    .attr('offset', (d) => d.offset)
+    .attr('stop-color', (d) => d.color)
+}
+
+// ==============
+// === Update ===
+// ==============
+
+onMounted(() => {
+  emit('update:preprocessor', 'Standard.Visualization.Histogram', 'process_to_json_text')
+})
+
+onMounted(() => queueMicrotask(redrawData))
+watch([width, height, maxY], () => queueMicrotask(redrawData))
+watchEffect(redrawData)
+
+watch([width, height], () => queueMicrotask(endBrushing))
```

Comment:
we want to remove brushing on resize even though width and height aren't dependencies. i guess it might be possible to resize brushing instead of removing it? :thinking: